apps/test-app/frontend/src/components/tree-widget/StatelessTree.tsx

```diff
@@ -88,14 +91,12 @@ function Tree({ imodel, imodelAccess, height, width }: { imodel: IModelConnectio
     [getHierarchyLevelConfiguration],
   );
   const propertiesSource = useMemo<(() => Promise<PresentationInstanceFilterPropertiesSource>) | undefined>(() => {
-    if (!hierarchyProvider || !filteringOptions) {
+    if (!filteringOptions) {
       return undefined;
     }
 
     return async () => {
-      const inputKeysIterator = hierarchyProvider.getNodeInstanceKeys({
-        parentNode: filteringOptions.options.hierarchyNode,
-      });
+      const inputKeysIterator = filteringOptions.options.getInstanceKeysIterator();
```

Comment:
it seems `filteringOptions` needs a new name, since it stores not only filtering options anymore

Reply:
Left `filteringOptions` name as it is but changed `filteringOptions.options` -> `filteringOptions.hierarchyDetails`. I think `filteringOptions` is good enough name in this situation because in the context of the component it stores options needed for filtering: node id and hierarchy details.